src/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -262,12 +262,20 @@ public BAMIndex getIndex() {
     public void setEagerDecode(final boolean desired) { this.eagerDecode = desired; }
     
     public void close() {
+    	if (mCompressedInputStream != null) {
+    		try {
+				mCompressedInputStream.close();
+			} catch (IOException e) {
+				throw new SAMException("Exception closing compressed input stream.", e);
+			}
+    	}
```

Comment:
probably a comment for all your code, but would it be possible to convert every tab to four spaces to match the style of the rest of the code?